Source/WebKit/WebProcess/cocoa/RemoteRealtimeMediaSourceProxy.cpp

```diff
@@ -169,6 +169,13 @@ void RemoteRealtimeMediaSourceProxy::whenReady(CompletionHandler<void(WebCore::C
     if (m_isReady)
         return callback(WebCore::CaptureSourceError(m_failureReason));
 
+    if (m_callback) {
+        callback = [firstCallback = std::exchange(m_callback, { }), secondCallback = WTFMove(callback)] (auto&& error) mutable {
```

Comment:
This is hard to read and understand, it looks like you can only deal with two Callbacks maximum.

I think it would be clearer if they were name previousCallbacks and newCallback respectively